core/src/main/java/org/akaza/openclinica/web/filter/ApiSecurityFilter.java

```diff
@@ -116,12 +116,22 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 
                             if (userType.equals(UserType.SYSTEM.getName())){
                                 String clientId = decodedToken.get("clientId").toString();
+                                HashMap<String, String> userAccountToCreate = null;
```

Comment:
this method is getting really big. Is it possible to refactor this code into a new method? I was thinking something like this...
Map<String, String> getUserAccountToCreate()...where all the client conditionals can be handled.